### ISSUE TITLE
“Bad From header”

### ISSUE THREAD

**Issue by qqilihq** [64characters/Telephone#399]
One of my colleagues updated his Mac to macOS 10.13. He's experiencing the following issue now when trying to make calls. Incoming calls are working just fine.

![telefon2](https://user-images.githubusercontent.com/2172260/32370627-2221f0b6-c08e-11e7-8ef1-98403171f42a.png)

Here's a snippet of the logfile in `~/Library/Containers/com.tlphn.Telephone/Data/Library/Application Support/com.tlphn.Telephone/Telephone.log`:

```
11:28:50.390   pjsua_core.c  .pjsua version 2.5.5 for Darwin-17.0/x86_64 initialized
11:28:50.530    pjsua_acc.c !....XXXXXX XXXXXX <sip:XXXXXXXXXXXXXXx@XXX.XXXXXXXX.XX>: registration success, status=200 (OK), will re-register in 90 seconds
11:28:56.876 PJSUAOnCallSta  ......Call 0 state changed to CALLING
11:28:57.165 PJSUAOnCallSta  .....Call 0 is DISCONNECTED [reason = 400 (Bad From header)]
11:29:50.333 PJSUAOnCallSta  ......Call 1 state changed to CALLING
11:29:50.721 PJSUAOnCallSta  .....Call 1 is DISCONNECTED [reason = 400 (Bad From header)]
```

On my own Mac with macOS 10.12.6 everything is working just fine (which should rule out the possibility, that the issue is called by our SIP provider).

Any ideas?

[minor side note: I was wondering why the Telephone.log was empty, and just accidentally discovered, that the Telephone.log only gets on application quit. Could probably be improved to avoid confusion?]

**Comment by qqilihq** [comment]
[Update] Our initial assumption of macOS 10.13 being the culprit was obviously wrong. On another Mac running 10.13 everything works fine.

Unfortunately, we were still not able to solve the original issue. Any suggestions welcome!

**Comment by eofster** [comment]
Please check if the following fields contain some potentially unacceptable characters. Maybe just try to recreate the whole account, if you haven’t already.

* Full Name
* Username
* Domain
* SIP Address (on the advanced account settings)

Regarding the log file: this asynchronous writing to file happens for performance reasons. Unfortunately, I can’t control it, that’s how PJSIP (SIP library that Telephone uses) is implemented.

**Comment by eofster** [comment]
By the way, the error is returned by the server. So to tell exactly why the server rejects the response, you could also check the log on the server, if you have control over it, of course.

**Comment by kkkrist** [comment]
Same problem here. I'm using an easybell Germany "Call Basic" account (sip.easybell.de). Every time I try to make a call it's cancelled and I get a "Bad From header" error. It started some days / 1-2 weeks ago and was around the time I upgraded to High Sierra. First problem in years, has been working flawlessly before (I'm a patron).

**Comment by kkkrist** [comment]
Some more testing and capturing is leading me to believe that this is indeed not a problem with Telephone but with Easybell's infrastructure.

I get the same error when I try to make an outgoing call with the same Easybell account using X-Lite on the mac and on Windows (albeit using Virtualbox) too. What works however is using the built-in dialer app of my Android phone.

So I set up a transparent proxy to capture the SIP session of my Android phone. The only difference I can spot with my limited SIP knowledge is that it uses a short numeric string as tag in the from header, whereas Telephone uses a long alphanumeric one and X-lite and short alphanumeric one.

Telephone
`From: "My Name" <sip:004900000000000@sip.easybell.de>;tag=HlSRZvsfIznED4UkVIs2NgzaxdjevETc`

X-Lite on macOS
`From: "My Name"<sip:004900000000000@sip.easybell.de>;tag=f9001d1c`

X-Lite on Windows
`From: "My Name"<sip:004900000000000@sip.easybell.de>;tag=f0659811`

Built-in Android dialer
`From: "My Number" <sip:004900000000000@sip.easybell.de>;tag=2865679377`

I will take this to the Easybell support now.

**Comment by kkkrist** [comment]
Got a reply from an Easybell support rep today claiming "My Name" as display name portion in the From header is invalid. He's obviously wrong, the [respective RFC](https://tools.ietf.org/html/rfc3261#section-20.20) even has according examples.

This got me curious enough to do some testing with different display names. I found out that Easybell's sip servers don't like whitespace inside the display name portion anymore (I could swear it worked for a long time until some days ago). 

According to the RFC whitespace is perfectly valid if the display name is denominated with quotes though (as Telephone does). So this is clearly a problem on Easybell's end.

TL;DR

As a workaround, remove any whitespace from the "full name" field in Telephone's account settings or don't set a name at all to make outgoing calls work with Easybell.

Let's hope they'll fix this soon. 

Issue can be closed I guess :-)

**Comment by eofster** [comment]
Thank’s a lot for the investigation!
